Sources/Ink/Internal/Link.swift

```diff
@@ -19,20 +20,35 @@ internal struct Link: Fragment {
 
         if reader.currentCharacter == "(" {
             reader.advanceIndex()
-            let url = try reader.read(until: ")")
-            return Link(target: .url(url), text: text)
+            let url = try reader.readCharacters(matching: \.isLegalInURL)
+
+            guard !reader.didReachEnd else { throw Reader.Error() }
+            var titleText: Substring? = nil
+            if reader.currentCharacter.isSameLineWhitespace {
+                try reader.readWhitespaces()
+                try reader.read("\"")
+                titleText = try reader.read(until: "\"")
+            }
+            try reader.read(")")
+            return Link(target: .url(url), text: text, title: titleText)
```

Comment:
```suggestion
            return Link(target: .url(url ?? ""), text: text, title: titleText)
```

Needed since `url` is now optional.